edk2toollib/uefi/edk2/path_utilities.py

```diff
@@ -100,17 +100,31 @@ def __init__(self, ws: os.PathLike, package_path_list: Iterable[os.PathLike],
             package_path_packages[package_path] = \
                 [Path(p).parent for p in package_path.glob('**/*.dec')]
 
+        ignore_nested_packages = False
+        if "STUART_IGNORE_EDK_NESTED_PACKAGES" in os.environ and \
+            os.environ["STUART_IGNORE_EDK_NESTED_PACKAGES"].strip().lower() == \
+                "true":
+            ignore_nested_packages = True
+
         for package_path, packages in package_path_packages.items():
             for i, package in enumerate(packages):
                 for j in range(i + 1, len(packages)):
                     comp_package = packages[j]
-
                     if (package.is_relative_to(comp_package)
                             or comp_package.is_relative_to(package)):
-                        raise Exception(
-                            f"Nested packages not allowed. The packages "
-                            f"[{str(package)}] and [{str(comp_package)}] are "
-                            f"nested")
+                        if ignore_nested_packages:
+                            self.logger.log(
+                                logging.WARNING,
+                                f"Nested packages not allowed. The packages "
+                                f"[{str(package)}] and [{str(comp_package)}] are "
+                                f"nested.")
```

Comment:
Should this printed text also indicate that user has environment variable set to ignore this error.  

Should it also mention that some pytools features may not work as expected due to ignoring this requirement?

Reply:
Added in PR commit https://github.com/tianocore/edk2-pytool-library/pull/199/commits/faf600ea5c0691f6f30c5ffdc9650750c245a90c